tests/testthat/test-impact.R

```diff
@@ -123,4 +123,141 @@ test_that("test if vimpact functions are working as expected for central estimat
 
 })
 
+test_that("impact calculation by year of vaccination country perspective", {
+  raw_impact <- data_frame(
+    country = c(rep("ETH", 5), rep("PAK", 5)),
+    year = rep(2001:2005, 2),
+    age = rep(0, 10),
+    value = c(234, 456, 345, 234, 345, 934, 567, 876, 675, 456),
+    burden_outcome = rep("deaths", 10)
+  )
+
+  fvps <- data_frame(
+    vaccine = rep("HepB", 10),
+    activity_type = rep("routine", 10),
+    country = c(rep("ETH", 5), rep("PAK", 5)),
+    year = rep(2001:2005, 2),
+    age = rep(0, 10),
+    fvps = c(34, 54, 34, 54, 23, 65, 78, 98, 78, 98),
+    disease = rep("HepB", 10)
+  )
+
+  impact <- impact_by_year_of_vaccination_country_perspective(
+    raw_impact, fvps, "routine", 2000:2030)
+  expect_equal(colnames(impact),
+              c("country", "burden_outcome", "value", "fvps", "impact_ratio"))
+  ## One row for each country, burden_outcome combination
+  expect_equal(nrow(impact), 2)
+  expect_equal(unique(impact$country), c("ETH", "PAK"))
+  expect_equal(unique(impact$burden_outcome), "deaths")
+  expect_equal(impact$value, c(1614, 3508))
+  expect_equal(impact$fvps, c(199, 417))
+  expect_equal(impact$impact_ratio, c(8.1, 8.4), tolerance = 1.e-1)
+
+  ## Impact for routine filters on birth cohorts in range of
+  ## year of vaccinations - min age fvps
+  fvps$age <- rep(1, 10)
+  impact <- impact_by_year_of_vaccination_country_perspective(
```

Comment:
I'm curious about this edge case as in calculating the impact ratio the impact is summed from 2 values, from the 2004 and 2005 data but the fvps used are only the 2005 ones - which gives a much higher impact_ratio than the calculation above - but maybe that is intended?